File: backend/src/main/java/org/kmurygin/healthycarbs/offers/mealPlanTemplate/MealPlanTemplateDTO.java
package org.kmurygin.healthycarbs.offers.mealPlanTemplate;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.util.List;

public record MealPlanTemplateDTO(
        Long id,
        @NotNull
        @Size(min = 3, max = 60)
        String name,
        String description,
        List<Long> dayIds,
        Double totalCalories,
        Double totalCarbs,
        Double totalProtein,
        Double totalFat
) {
}

File: frontend/src/app/core/models/dto/meal-plan-template.dto.ts
export interface MealPlanTemplateDto {
  readonly id: number;
  readonly name: string;
  readonly description: string | null;
  readonly totalCalories: number | null;
  readonly totalCarbs: number | null;
  readonly totalProtein: number | null;
  readonly totalFat: number | null;
}
